portfolio/src/test/java/com/google/sps/servlets/CommentServletTest.java

```diff
@@ -0,0 +1,136 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+
+/**
+ * Unit tests for the Course class.
+ */
+ @RunWith(JUnit4.class)
+public final class CommentServletTest {
+  private final LocalServiceTestHelper helper = 
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private static final CommentServlet commentServlet = new CommentServlet();
+  private HttpServletRequest mockRequest;
+  private HttpServletResponse mockResponse;
+  private String tokenId;
+
+  @Before
+  public void setUp() throws IOException, ServletException {
+    tokenId = getTokenId();
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    helper.setUp();
+    MockitoAnnotations.initMocks(this);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testGet() throws IOException {
+    String data = Comment.convertToJson(generateRandomComments());
+    StringWriter stringWriter = new StringWriter();
+    stringWriter.write(data);
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(mockResponse.getWriter()).thenReturn(writer);
+
+    commentServlet.doGet(mockRequest, mockResponse);
+    Assert.assertTrue(stringWriter.toString().contains(data));
+  }
+
+  private List<Comment> generateRandomComments() {
+    List<Comment> comments = new ArrayList<>();
+    for (int i = 1; i <= 5; i++) {
+      comments.add(new Comment(
+        Long.valueOf(i), 
+        System.currentTimeMillis(), 
+        "user" + i, 
+        "username" + i, 
+        "hello, this is a comment #" + i, 
+        0, 
+        Comment.Status.PUBLIC
+      ));
+    }
+    return comments;
+  }
+
+  @Test
+  public void testPost() throws IOException, ServletException {
+    testGet();
+    Mockito.when(mockRequest.getContentType()).thenReturn("multipart/");
+    Mockito.when(mockRequest.getParameter("id")).thenReturn(tokenId);
+
+    Collection<Part> parts = RequestDataTest.mockRequestParts();
+    Mockito.when(mockRequest.getParts()).thenReturn(parts);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Assert.assertEquals(0, datastore.prepare(new Query("Comment")).countEntities());
+    
+    commentServlet.doPost(mockRequest, mockResponse);
+
+    Assert.assertEquals(1, datastore.prepare(new Query("Comment")).countEntities());
+  }
+
+  @Test
+  public void testDelete() throws IOException, ServletException {
+    testGet();
+    testPost();
+    Mockito.when(mockRequest.getParameter("id")).thenReturn(tokenId);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Filter deletedOnlyFilter = new FilterPredicate(
+      "status", FilterOperator.EQUAL, Comment.Status.DELETED.name()
+    );
+    Query query = new Query("Comment").setFilter(deletedOnlyFilter);
+
+    Assert.assertEquals(0, datastore.prepare(query).countEntities());
```

Comment:
Only test one thing at a time.  In general you don't want to have multiple assertions in a test unless you're
A) Testing a critical user journey that is a series of commands
B) Testing different properties of an item where the only way to test the different properties is individual assertions.

Here you are testing two different functionalities in the same test which is wrong.  Should be two different tests.

Also IMO this line doesn't need to be tested at all.  You aren't testing the deletion here, all you're testing is that datastore can run queries properly which you already know.